src/routes/api/users/[userId]/demos/[demoId]/index.ts

```diff
@@ -81,35 +92,41 @@ export const POST: RequestHandler = async (requestEvent: RequestEvent) => {
   }
 };
 
-export const PATCH = async (requestEvent: RequestEvent) => {
-  try {
-    const isFile = ['backgroundPoster', 'brandLogo', 'sipImage1', 'sipImage2', 'sipImage3', 'sipImage4'];
-    const formData = await requestEvent.request.formData();
-    const form = await Promise.all(
-      [...formData.entries()].map(async (field) => {
-        if (isFile.includes(field[0])) field[1] = await toData(field[1]);
+export const PUT = async (requestEvent: RequestEvent) => {
+  const formData = await requestEvent.request.formData();
+  const form = await Promise.all(
+    [...formData.entries()].map(async (field) => {
+      //Check if its a file or not
+      field[1] = field[1] instanceof Object ? await toData(field[1]) : JSON.parse(field[1]);
+      return field;
+    })
+  );
 
-        if (field[1] == 'true') field[1] = true;
-        if (field[1] == 'false') field[1] = false;
+  try {
+    const db = await getDB();
+    const demo = await findDemo(db, requestEvent.params.demoId, requestEvent.params.userId);
 
-        return field;
-      })
-    );
+    form.forEach((curr) => (demo[curr[0]] = curr[1]));
+    await db.persistAndFlush(demo as Demo);
 
-    const db = await MikroORM.init({
-      ...config,
-      ...{ entities: [entity.Session, entity.User, entity.Data, entity.Demo] }
-    }).then((r) => r.em.fork());
+    return { status: 200, body: { ...generateDemo(demo), uuid: demo.uuid } };
+  } catch (e) {
+    return { status: 500 };
+  }
+};
 
-    const demo = await db.findOne(
-      entity.Demo,
-      { uuid: requestEvent.params.demoId, user: { uuid: requestEvent.params.userId } },
-      { fields, strategy: LoadStrategy.JOINED }
-    );
+export const PATCH = async (requestEvent: RequestEvent) => {
+  const body = await requestEvent.request.json();
 
-    form.forEach((curr) => (demo[curr[0]] = curr[1]));
+  try {
+    const db = await getDB();
+    let demo = await findDemo(db, requestEvent.params.demoId, requestEvent.params.userId);
 
-    await db.persistAndFlush(demo);
+    demo = Object.keys(body).reduce((acc, curr) => {
+      acc[curr] = body[curr];
+      return acc;
+    }, demo);
+    await db.persistAndFlush(demo as Demo);
```

Comment:
For consistency, please use promise chaining instead of await. In my opinion, it would also be simpler to understand.

Reply:
I have to disagree with you here. IMO await/async is the way to go. to avoid extra indentation and declare extra callback methods which enforce use to add more parenthesis and arrows and other operation symbols. This clearly make the code harder to read.